api/lib/domain/usecases/authenticate-external-user.js

```diff
@@ -47,7 +47,8 @@ async function authenticateExternalUser({
     });
 
     if (userFromCredentials.shouldChangePassword) {
-      throw new UserShouldChangePasswordError();
+      const passwordResetToken = tokenService.createPasswordResetToken(userFromCredentials.id);
+      throw new UserShouldChangePasswordError(undefined, passwordResetToken);
```

Comment:
idem: utiliser un objet comme paramètre au lieu de 2 paramètres

```suggestion
throw new UserShouldChangePasswordError({ passwordResetToken });
```